### DISCUSSION TITLE
manual accuracy check FAILS

### DISCUSSION THREAD

**Discussion by AnirudhG07** [mrdbourke/pytorch-deep-learning#769] [Q&A]
hello, I need help. I am currently learning basics of pytorch and so I myself tried MNIST numbers data.
So my accuracy came out good. 97% around but when I check accuracy manually by checking calling each data element and comparing actual to predicted number, the accuracy I am getting is <10%

I am sharing codes of checking and some other areas you might need to see.
<img width="1083" alt="image" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/146579014/cfe30eb4-e71d-4c85-8b38-cc1cd695811e">

<img width="1083" alt="image" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/146579014/73893a5f-1383-4834-b23e-fb1676a6bd0c">

<img width="1083" alt="image" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/146579014/1af51fe9-b30f-4153-a2c5-561ea392d543">

<img width="1083" alt="image" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/146579014/97e8ad13-8159-4688-b38d-07e339e7f8e6">
Pls HELP!

**Comment by sametucuncu** [comment]
You may use test_loader to iterate through examples for evaluation.
Try to put some print statements between lines to see if you get something wrong. (Especially after scores = model(data) and the one below.)

**Reply by AnirudhG07** [reply]
Its giving 38% accuracy with data_loader. and decreases every time I run it again and again. while training, the model is evaluating with 97% accuracy.
I gave it an image of mine and it is giving random incorrect answers. 
Tell me how to fix that please
